src/Features/Core/Portable/ValidateFormatString/AbstractValidateFormatStringDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,295 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Threading;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.LanguageServices;
+
+namespace Microsoft.CodeAnalysis.ValidateFormatString
+{
+    internal abstract class AbstractValidateFormatStringDiagnosticAnalyzer<TSyntaxKind> :
+        DiagnosticAnalyzer
```

Comment:
I'm hesitant to add any new diagnostic that doesn't have a way to be disabled.  We've even been asked for ways to disable fading of unused usings, etc.